ecommerce/views/v0/__init__.py

```diff
@@ -173,6 +175,71 @@ def create(self, request, *args, **kwargs):
         )
 
 
+def generate_checkout_payload(request):
+    try:
+        basket = Basket.objects.filter(user=request.user).get()
+    except ObjectDoesNotExist:
+        return Response("No basket", status=status.HTTP_406_NOT_ACCEPTABLE)
+
+    discounts = BasketDiscount.objects.filter(redeemed_basket=basket).all()
+
+    total = 0
+
+    order = PendingOrder(total_price_paid=0, purchaser=request.user)
+    order.save()
+
+    ip = get_client_ip(request)[0]
+
+    gateway_order = GatewayOrder(
+        username=request.user.username,
+        ip_address=ip,
+        reference=order.reference_number,
+        items=[],
+    )
+
+    for lineitem in basket.basket_items.all():
+        line_item_price = lineitem.product.price
+
+        if len(discounts) > 0:
+            for discount in discounts:
+                discount_cls = DiscountType.for_discount(discount.discount_type)
+                discounted_price = discount_cls.get_product_price(lineitem.product)
+                total += discounted_price
+                line_item_price = discounted_price
+        else:
+            total += lineitem.product.price
+
+        product_version = Version.objects.get_for_object(lineitem.product).first()
+        Line(order=order, product_version=product_version, quantity=1).save()
+        gateway_order.items.append(
+            GatewayCartItem(
+                code=product_version.field_dict["content_type_id"],
+                name=product_version.field_dict["description"],
+                quantity=1,
+                sku=f"{product_version.field_dict['content_type_id']}-{product_version.field_dict['object_id']}",
+                unitprice=line_item_price,
+                taxable=0,
+            )
+        )
+
+    order.total_price_paid = total
+    order.save()
```

Comment:
Since we've now created a pending order, we've consumed the items in the cart - so we should clear the cart out.